troubleshooting.md

```diff
@@ -118,28 +118,38 @@ troubleshooting guide.
 This guide will provide an example of troubleshooting a scenario where
 an image is not written to Ceph in Glance. However, the steps are also
 applicable Cinder or Nova being unable to write to Ceph. We assume
-that the pods is not crashlooping as described in the section above.
+that the pods might not be in crashlooping status as described in the
+section above.
 
-1. Confirm the pod has the Ceph configuration files
+1. Confirm the `Pod` has the Ceph configuration files
 
 The [Ceph integration documentation](ceph.md) describes using
 `ExtraMounts` to place files like `ceph.client.openstack.keyring`
-and `ceph.conf` in `/etc/ceph` in each pod which needs to connect
-to Ceph. Connect to the pod with `oc rsh` and ensure that both of
-these files are present.
+and `ceph.conf` in `/etc/ceph` in each `Pod` which needs to connect
+to Ceph.
+Run `oc describe <glance_pod>` and look for both `Mounts` and `Volumes` list:
+if the Ceph related bits have been properly propagated, there must be an entry
+pointing to `/etc/ceph` which is populated by the referenced Ceph `Secret`.
+If the Ceph files are missing, check the syntax of the `ExtraMounts` section of
+the spec in the `OpenStackControlPlane`. If the Glance `Pod` is in a
+`CrashLoopBackOff` status, it is required to build a debug `Pod` and run it as
+described in the [glance-operator troubleshooting
+guide](https://github.com/openstack-k8s-operators/glance-operator/blob/main/docs/dev/troubleshooting.md)
 
-If the Ceph files are missing, check the syntax of
-the `ExtraMounts` section of the spec in the `OpenStackControlPlane`.
+Run a debug `Pod` with the following command:
+
+```
+oc debug pod/glance-default-external-api-0 --keep-labels=true
+```
 
 2. Confirm the pod can reach the Ceph cluster
 
-The Ceph cluster should be reachable from the Storage network as
-described in the [networking document](networking.md). The IPs and
-ports of the Ceph cluster's monitors should be in
-`/etc/ceph.conf`. From within the pod, read this and try to ping the
-IP as well as connect to the port.
+The Ceph cluster should be reachable from the Storage network as described in
+the [networking document](networking.md). The IPs and ports of the Ceph
+cluster's monitors should be in `/etc/ceph.conf`. From within the pod, read
+this and try to reach the IP as well as connect to the port.
 ```
-$ oc rsh -c glance-api glance-default-external-api-0
+$ oc debug pod/glance-default-external-api-0
```

Comment:
I think it should be `oc debug --container glance-api pod/glance-default-external-api-0` instead.

When I use `--container glance-api` then I can see the relevant files.
```
[zuul@controller-0 ~]$ oc debug --container glance-api pod/glance-3d293-default-external-api-0 
Warning: would violate PodSecurity "restricted:v1.24": allowPrivilegeEscalation != false (containers "glance-log", "glance-httpd", "glance-api" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (containers "glance-log", "glance-httpd", "glance-api" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or containers "glance-log", "glance-httpd", "glance-api" must set securityContext.runAsNonRoot=true), runAsUser=0 (containers "glance-log", "glance-httpd", "glance-api" must not set runAsUser=0), seccompProfile (pod or containers "glance-log", "glance-httpd", "glance-api" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
Starting pod/glance-3d293-default-external-api-0-debug-sbtgs, command was: /usr/bin/dumb-init --single-child -- /bin/bash -c /usr/local/bin/kolla_set_configs && /usr/local/bin/kolla_start
Pod IP: 192.168.23.132
If you don't see a command prompt, try pressing enter.
sh-5.1# ls -l /etc/ceph/
total 0
lrwxrwxrwx. 1 root root 36 Jul 29 13:09 ceph.client.openstack.keyring -> ..data/ceph.client.openstack.keyring
lrwxrwxrwx. 1 root root 16 Jul 29 13:09 ceph.conf -> ..data/ceph.conf
sh-5.1# 
```
But without it, then I cannot see the ceph.conf in my debug pod.
```
[zuul@controller-0 ~]$ oc get pods | grep glance
glance-3d293-default-external-api-0                               3/3     Running     0             47h
glance-3d293-default-external-api-1                               3/3     Running     0             47h
glance-3d293-default-external-api-2                               3/3     Running     0             47h
glance-3d293-default-internal-api-0                               3/3     Running     0             47h
glance-3d293-default-internal-api-1                               3/3     Running     0             47h
glance-3d293-default-internal-api-2                               3/3     Running     0             47h
glance-dbpurge-28700641-6prqz                                     0/1     Completed   0             2d13h
glance-dbpurge-28702081-g5xlx                                     0/1     Completed   0             37h
glance-dbpurge-28703521-rd5z9                                     0/1     Completed   0             13h
[zuul@controller-0 ~]$ oc debug pod/glance-3d293-default-external-api-0
Defaulting container name to glance-log.
Use 'oc describe pod/glance-3d293-default-external-api-0-debug-tqjjq -n openstack' to see all of the containers in this pod.

Warning: would violate PodSecurity "restricted:v1.24": allowPrivilegeEscalation != false (containers "glance-log", "glance-httpd", "glance-api" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (containers "glance-log", "glance-httpd", "glance-api" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or containers "glance-log", "glance-httpd", "glance-api" must set securityContext.runAsNonRoot=true), runAsUser=0 (containers "glance-log", "glance-httpd", "glance-api" must not set runAsUser=0), seccompProfile (pod or containers "glance-log", "glance-httpd", "glance-api" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
Starting pod/glance-3d293-default-external-api-0-debug-tqjjq, command was: /usr/bin/dumb-init --single-child -- /bin/sh -c /usr/bin/tail -n+1 -F /var/log/glance/glance-3d293-default-external.log 2>/dev/null
Pod IP: 192.168.23.131
If you don't see a command prompt, try pressing enter.
sh-5.1# ls -l /etc/ceph/
total 4
-rw-r--r--. 1 root root 92 Apr 29 21:07 rbdmap
sh-5.1# 
```

Reply:
+1 good catch! Updating accordingly!

Reply:
Done (thank you!)